server/src-lib/Hasura/RQL/DDL/Schema/Table.hs

```diff
@@ -189,18 +189,35 @@ processTableChanges ti tableDiff = do
         if | oColName /= nColName -> do
                renameColInCatalog oColName nColName tn ti
                return True
+
            | oColTy /= nColTy -> do
                let colId   = SOTableObj tn $ TOCol oColName
-                   depObjs = getDependentObjsWith (== "on_type") sc colId
-               unless (null depObjs) $ throw400 DependencyError $
+                   typeDepObjs = getDependentObjsWith (== "on_type") sc colId
```

Comment:
Assigning a particular meaning to a string value like this makes me nervous. Can we use a sum type here, instead? Something like

```haskell
data ColumnDependencyReason = DependentOnType | DependentOnSessionVariable
```

except maybe with better names than I can come up with based on my understanding. :)